netbox/users/models.py

```diff
@@ -227,6 +237,24 @@ def is_expired(self):
             return False
         return True
 
+    def validate_client_ip(self, raw_ip_address):
+        """
+        Checks that an ip address falls within the allowed ips.
+        """
+        if not self.allowed_ips:
+            return True
+
+        try:
+            ip_address = ipaddress.ip_address(raw_ip_address)
+        except ValueError:
+            raise ValidationError(f"{raw_ip_address} is an invalid IP address")
```

Comment:
I feel like we should return a 400 Bad Request response if the IP address is invalid. (We can handle this under `authenticate()` by catching ValidationError. Otherwise, raising `ValidationError` (which AFAICT won't be caught by DRF) is just going to trigger a 500 so we should figure out something a bit cleaner.

Reply:
Not sure how to do this.. Any example code that accomplish this ?

Reply:
We should be able to catch the `ValidationError` and return a [`HttpResponseBadRequest`](https://docs.djangoproject.com/en/3.2/ref/request-response/#django.http.HttpResponseBadRequest) response, with an error message indicating the invalid value.

Reply:
Hm.. not easy :-)
Do I need to extend the ExceptionHandlingMiddleware to catch the error and generate a 400 ?
a simple `return HttpResponseBadRequest(f"ValidationErrorInfo: {ValidationErrorInfo.message}")` just throws another error that the return of authenticate() expects 2 values, but does not return the 404 error to the client

Reply:
Found it :
```
except ValidationError as ValidationErrorInfo:
    raise exceptions.ValidationError(f"The Source IP in the HTTP Header {http_header} has a ValidationError: {ValidationErrorInfo.message}")
```

This raises a `HTTP/1.1 400 Bad Request`

Reply:
@jeremystretch Does this solution seems good ?